cmd/keymasterd/certgen.go

```diff
@@ -302,6 +334,17 @@ func (state *RuntimeState) postAuthX509CertHandler(
 			logger.Printf("Cannot parse public key")
 			return
 		}
+		validKey, err := certgen.ValidatePublicKeyStrength(userPub)
+		if err != nil {
+			logger.Println(err)
+			state.writeFailureResponse(w, r, http.StatusInternalServerError, "")
+			return
+		}
+		if !validKey {
+			state.writeFailureResponse(w, r, http.StatusBadRequest, "Invalid File, Check Key strength/key type")
```

Comment:
it would be nice to tell the user what the threshold specifically (if it's exponent length, or, key size, or what, so they can be sure to fix it)